tmt/checks/__init__.py

```diff
@@ -83,6 +84,19 @@ def from_spec(cls, spec: str) -> 'CheckEvent':
             raise tmt.utils.SpecificationError(f"Invalid test check event '{spec}'.")
 
 
+class CheckResultInterpret(enum.Enum):
+    INFO = 'info'
+    RESPECT = 'respect'
+    XFAIL = 'xfail'
+
+    @classmethod
+    def from_spec(cls, spec: str) -> 'CheckResultInterpret':
+        try:
+            return CheckResultInterpret(spec)
+        except ValueError:
+            raise ValueError(f"Invalid check result interpretation '{spec}'.")
```

Comment:
Does it make sense to use `raise ... from ...` here?

```
except ValueError as error:
    raise ValueError(f"Invalid check result interpretation '{spec}'.") from error
```

Reply:
Not necessarily, since `spec` is shown, it should be fairly clear why it failed the validation.

Reply:
Good discussion point though. It's not generally being used here, the [B904](https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/#raise-without-from-inside-except-b904) is marked in 'ignore'.  

So, if we want to adhere to this rule, there could either be a pull request with enabled check and modified all the existing violations ... or the existing exceptions could get `noqa` so the rule can be removed from ignore list and any new code would comply.

Reply:
Added in https://github.com/teemtee/tmt/pull/3239/commits/345bde830384b610d0d4bccd395b7f0a00aa5b26 
Created https://github.com/teemtee/tmt/issues/3308 to not forget about this.